### PR TITLE
Add nncf task

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
* add supporting NNCF compression
* Split ote task`OTESegmentationTask` to `OTESegmentationInferenceTask`, `OTESegmentationTrainingTask` and  `OTESegmentationNNCFTask`

Compression result:

  | mDice | mDice (on train) | FPS (async) | SpeedUp
-- | -- | -- | -- | --
orig (fp32) | 0.878 | 0.9282 | 28.77 |  
**NNCF (int8)** | **0.862** | **0.9182** | **53.82** | **1.8707**
POT (int8) |   |   | 53.16 | 1.8478

### CODE REVIEW

File: mmseg/apis/ote/apis/segmentation/train_task.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        labels = {label.name: label.color.rgb_tuple for label in self._labels}
+        model_info = {'model': self._model.state_dict(), 'config': hyperparams_str, 'labels': labels, 'VERSION': 1}
+
+        buffer = io.BytesIO()
+        torch.save(model_info, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        output_model.set_data("label_schema.json", label_schema_to_bytes(self._task_environment.label_schema))
+
+
+    def cancel_training(self):
```
Comment (evgeny-izutov) [inline mmseg/apis/ote/apis/segmentation/train_task.py:159]: The method duplicate the existing one in the parent class.

File: mmseg/apis/ote/sample/sample_nncf.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+from ote_sdk.entities.resultset import ResultSetEntity
+from ote_sdk.entities.subset import Subset
+from ote_sdk.usecases.tasks.interfaces.export_interface import ExportType
+from ote_sdk.usecases.tasks.interfaces.optimization_interface import OptimizationType
+from ote_sdk.entities.task_environment import TaskEnvironment
+
+from mmseg.apis.ote.apis.segmentation.ote_utils import get_task_class
+from mmseg.apis.ote.extension.datasets.mmdataset import load_dataset_items
+
+from sc_sdk.entities.datasets import NullDataset
```
Comment (evgeny-izutov) [inline mmseg/apis/ote/sample/sample_nncf.py]: We should exclude all references on sc_sdk.

File: mmseg/apis/ote/apis/segmentation/train_task.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply mmseg/apis/ote/apis/segmentation/train_task.py:159]: removed from inference_task

**Review by alexsu52 — APPROVED** [review]
It looks good to me.

**Review by evgeny-izutov — APPROVED** [review]

**Pull request merged**